[git-metadata] Update command limitations (#1303)
DataDog/datadog-ci: src/commands/git-metadata/README.md

#### Limitations

> This command will only report commits with a commit date within the last 30 days.

Reviewer: Do we want to say `within the last 30 days, up to 1,000 commits.` ?

Author: yes we do 🙃

> This command will only report commits in the current HEAD's history, with a commit date within the last 30 days and up to 1,000 commits.

Reviewer 2: ```suggestion
The `git-metadata` command only reports commits in the current HEAD's history with a commit date within the last 30 days. The command outputs a maximum of 1,000 commits.
```
